### PR TITLE
Take latest bogo test suite

### LINKED ISSUE #1158: Add support for latest BoGo updates
current output, running against unpatched google master (396625d50212143bc03517c9fdd21b1c965b3f45): https://gist.github.com/jbr/8d2a0d96719132f06666ff47544d8c89

it looks like there are unsupported flags on the shim as well as reported failures

### PR DESCRIPTION
**Pull request by ctz**
resolves #1158

### CODE REVIEW

File: rustls/src/error.rs [hunk 1]
```diff
@@ -225,6 +225,7 @@ pub enum PeerMisbehaved {
     UnsolicitedSctList,
     UnsolicitedServerHelloExtension,
     WrongGroupForKeyShare,
+    MissingPskModesExtension,
```
Comment (djc) [inline rustls/src/error.rs:228]: Keep this in alphabetic order?

**Review by djc — APPROVED** [review]

File: bogo/config.json [hunk 2]
```diff
@@ -33,6 +33,8 @@
     "DuplicateCertCompressionExt*": "",
     "TLS-ECH-*": "",
     "ALPS-*": "",
+    "Kyber*": "",
+    "*Kyber*": "",
```
Comment (cpu) [inline bogo/config.json:37]: Do we need both of these? Wouldn't `*Kyber*` match the same things as `Kyber*` or am I missing a detail?

File: rustls/src/error.rs [hunk 3]
```diff
@@ -225,7 +226,6 @@ pub enum PeerMisbehaved {
     UnsolicitedSctList,
     UnsolicitedServerHelloExtension,
     WrongGroupForKeyShare,
-    MissingPskModesExtension,
```
Comment (cpu) [inline rustls/src/error.rs]: Rebase artifact? The commit leads with `fixup!` and doesn't appear to be doing what it says on the tin.

Comment (cpu) [reply rustls/src/error.rs]: Aha, already caught. I was reviewing the stale branch :-)

File: bogo/fetch-and-build [hunk 4]
```diff
@@ -15,7 +15,7 @@ util/testresult
 EOF
 
 # fix on a tested point of rustls-testing branch
-COMMIT=3fea50a7cb55bb72e4a69d34015f1a6971e7934c
+COMMIT=b81c2d26009b6cc5159b7a9cbdddfa4ae79defea
```
[the remark this answers is not in the document]
Comment (ctz) [reply bogo/fetch-and-build:18]: This is now the tip of the `rustls-testing` branch, which has two commits additional to upstream.

File: bogo/config.json (hunk 2 above)
Comment (ctz) [reply bogo/config.json:37]: You're completely right -- will clean this up.

File: bogo/fetch-and-build (hunk 4 above)
[the remark this answers is not in the document]
Comment (cpu) [reply bogo/fetch-and-build:18]: Understood, thanks :)

**Pull request merged**